lib/bolt/node/ssh.rb

```diff
@@ -6,13 +6,27 @@
 module Bolt
   class SSH < Node
     def connect
-      options = { logger: @transport_logger,
-                  non_interactive: true }
+      options = {
+        logger: @transport_logger,
+        non_interactive: true
+      }
+
       options[:port] = @port if @port
       options[:password] = @password if @password
+      options[:verify_host_key] = if @insecure
+                                    Net::SSH::Verifiers::Lenient.new
+                                  else
+                                    Net::SSH::Verifiers::Secure.new
+                                  end
 
       @session = Net::SSH.start(@host, @user, options)
       @logger.debug { "Opened session" }
+    rescue SocketError, SystemCallError => e
+      raise Bolt::Node::Exception,
+            "Failed to connect to #{@uri}: #{e.message}"
+    rescue Net::SSH::HostKeyError => e
+      raise Bolt::Node::Exception,
+            "Host key verification failed for #{@uri}: #{e.message}"
```

Comment:
This should also handle authentication failures gracefully, e.g. Net::SSH::AuthenticationFailed.